debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -134,7 +139,7 @@ public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPoi
 
 					while (iterator.hasNext()) {
 						J9ObjectFieldOffset result = iterator.next();
-						if (result.getName().equals(nestingHierarchy[i])) {
```

Comment:
If I skip the for loop, I won't be able to find the nestedFieldOffset that I also need to update the dataStart by

Reply:
I meant:

```
for (...)
    clazz = nestedClassHierarchy[i];
    if (isArray(clazz) {
        //increment data start by index*stride
    } else {
        for (...)
    }
}
```
So you only need to skip that for loop for the `[#]` iteration because in that iteration you dont need the nestedfieldoffset. The following iteration will either be a nested field or nothing which is handled by the logic above